pkg/api/v20191231preview/openshiftcluster_validatestatic.go

```diff
@@ -188,6 +188,10 @@ func (sv openShiftClusterStaticValidator) validateServicePrincipalProfile(path s
 }
 
 func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np *NetworkProfile) error {
+	if np.PodCIDR == api.JoinCIDR || np.ServiceCIDR == api.JoinCIDR {
```

Comment:
It can't check the overlap with the different subnet mask.
We may want to use this https://pkg.go.dev/github.com/apparentlymart/go-cidr/cidr#VerifyNoOverlap.

Reply:
Just to add on top of Bitoku comments, the unit test cases for all the internalAPI versions can also use the podCIDR/serviceCIDR such that it covers the reserved ranges partially. For Eg. podCIDR: 100.64.0.0/18

Reply:
The requested changes completed. New approach https://stackoverflow.com/a/34729915 used